articles/discord-bot-node/index.md

```diff
@@ -0,0 +1,143 @@
+
+Make a folder for your project. You can call it anything. Once you're in that folder, type `npm install discord.js`. It will take a bit of time. You might see a few warnings, but that's perfectly normal. Make sure you have administrative privileges if you’re on Windows, and use `sudo` on Linux.
+
+![The npm install discord.js command](/engineering-education/discord-bot-node/discordbot8.png)
+
+
+### Getting Your Token
+
+The token for your bot is like a username and a password. It tells Discord which bot you want to control, and proves that you are the owner of the bot. **DO NOT GIVE AWAY YOUR DISCORD TOKEN!** I sometimes accidentally put my token on GitHub. Then I have to go through the trouble of resetting it. I'll show you how to hide it from GitHub in just a sec.
+
+![“Click Here To Reveal Your Token”](/engineering-education/discord-bot-node/discordbot5.png)
+
+If this project is on GitHub, you'll want to hide your token. Create a file called `token.txt`, and add it to your `.gitignore` file. You might want to put some random text in `token.txt` first, and then push to make sure it’s actually hidden. Once you’re sure you’ve done it correctly, put the token in your `token.txt` file. Now, we can start coding.
+
+Make a file called `index.js` and open it up in your text editor of choice. There's some basic boilerplate code that you'll need to get started.
+
+```javascript
+const Discord = require("discord.js"); // imports the discord library
+const fs = require("fs"); // imports the file io library
+
+const client = new Discord.Client(); // creates a discord client
+const token = fs.readFileSync("token.txt").toString(); // gets your token from the file
+
+client.once("ready", () => { // prints "Ready!" to the console once the bot is online
+	console.log("Ready!");
+});
+
+client.login(token); // starts the bot up
+```
+
+You can run this by using `node index.js`. Once you’ve done that it should look something like this:
+
+![The word “ready” appears in the command prompt](/engineering-education/discord-bot-node/discordbot9.png)
+
+## Creating Commands
+
+That's cool, but we want the bot to actually do something. The most common example is [ping-pong](https://www.youtube.com/watch?v=DEqrCI1018I), but we should do something a little more interesting. Let's generate random numbers. Whenever a user types in "?random", let's make the bot reply with a random number. We just need to add some more code above the `client.login()` call.
+
+```javascript
+client.on("message", message => { // runs whenever a message is sent
+    if (message.content === "?random") { // checks if the message says "?random"
+        const number = Math.random(); // generates a random number
+        message.channel.send(number.toString()); // sends a message to the channel with the number
+    }
+});
+```
+
+We did it! Albeit, in a [less than optimal way](http://www0.cs.ucl.ac.uk/staff/d.jones/GoodPracticeRNG.pdf).
```

Comment:
You need to give an explanation of this (TLDR at a minimum). The pdf you linked is just too long for them to make the intuitive leap that you are asking them to here.

Reply:
The link is there as a footnote. It's not really needed, and the point was to just say that the reader could create their own random function. I removed it.